### ISSUE TITLE
Add header and footer guidance

### ISSUE THREAD

**Issue by stephiescastle** [nasa-jpl/explorer-1#172]
### Description

Our documentation does not include any guidance or examples for the header and footer. We should add this to the Explorer-1 storybook, as users have been requesting this guidance, and it's also one of the final pieces to providing a full site template. 

Some notes to consider to help determine scope:

- Do we want to add both internal and external header/footer? Based on user requests, the external header/footer should be prioritized if we don't have time for both.
- What is our goal with providing the site header and footer? Is it to provide templates for building an external site with its own site navigation? Or is it to provide templates for building an external site that mimics a WWW sub site, such as: https://www.jpl.nasa.gov/go/astronomy-and-physics ? Note that in the sub-site, the header and footer are unchanged. Only the secondary navigation is custom.
- If the goal is to allow users to create WWW sub-sites, then we will also need to add guidance on secondary navigation and how it interacts with heroes on the page (scope increase)
- If the goal is to provide a customizable header and footer, note that in the external header:
  - There are no guidelines on where to put a site title since the header was originally designed for a single site: www.jpl.nasa.gov. Do we want to provide an alternative that includes a site title? We will need design guidance if so.
  - There is no "home" button or a way for users to navigate to the homepage. This wasn't an issue for JPL, as the tribrand logo was the "home" button. A simple solution, if design approves, is to add "Home" as the first item in the navigation. We could alternatively link the site title, if it gets added.
- How does this affect applications that only need the header and footer? Should we provide a separate resource for that?

**Comment by stephiescastle** [comment]
Results from a quick chat in our team meeting:
- We need both external and internal header/footer
- For the external header, we will probably need to offer a new option, as the header should be different from the public JPL website
- Footer: should relate to the site that it is on--its content does not need to be the same as the public JPL website (this is in response to the legacy guideline that would require the footer to match the footer from the public JPL website)
- We need clear documentation about how to use these external and internal components. A couple places this documentation could live:
  - All on Storybook
  - JPL's branding site could be the "accompanying documentation website" to Storybook, with more detail being covered there regarding internal and external site guidelines.
  - We may do a hybrid of the two approaches.

**Comment by Scotchester** [comment]
> We need clear documentation about how to use these external and internal components. A couple places this documentation could live:
> 
> * All on Storybook
> * JPL's branding site could be the "accompanying documentation website" to Storybook, with more detail being covered there regarding internal and external site guidelines.
> * We may do a hybrid of the two approaches.

I would personally prefer not to put policy-type content in Explorer 1's Storybook, and would prefer to keep that to usage guidelines for designers and developers.

**Comment by stephiescastle** [comment]
> I would personally prefer not to put policy-type content in Explorer 1's Storybook, and would prefer to keep that to usage guidelines for designers and developers.

Agreed. I'm thinking policy type documentation would go on the branding site, but I think we may need a clear indicator in Storybook differentiating the components. Perhaps just the organization in the sidebar is enough, or maybe even a visual badge, flagging something that needs special consideration (ex: ![image](https://img.shields.io/badge/External-red)).

**Comment by gekim84** [comment]
https://www.figma.com/file/4w0ycZrXntqiJU9K6zfyvZ/?node-id=0%3A10658

**Barebones Header and Footer**
![Generic Header](https://user-images.githubusercontent.com/44758056/163072892-ef245f53-1bef-4d37-9d5f-56e465acd0bf.jpg)
![Generic Footer](https://user-images.githubusercontent.com/44758056/163072886-762615ce-0765-4c04-a244-f0ff379a4df6.jpg)

**Example - Science and Technology**
![Top Hat - Science   Technology Example](https://user-images.githubusercontent.com/44758056/163072975-4e522ea1-b4c1-4378-946a-1bad362a1264.jpg)
![Footer Science   Technology](https://user-images.githubusercontent.com/44758056/163072967-7753f06c-34d1-4461-9a4f-9baef091ac97.jpg)

**Example of Hover States:** (Follows WCP styling except the down chevron arrows)
<img width="1278" alt="Screen Shot 2022-04-12 at 4 56 14 PM" src="https://user-images.githubusercontent.com/44758056/163073250-30215643-0be0-40b8-9048-c236908d68a3.png">

**Comment by gekim84** [comment]
submitted for design review

**Comment by gekim84** [comment]
### v9 - Uncertainty Hidden
![JPL - Asteroid Watch Page Desktop - v9 - Uncertainty Hidden](https://user-images.githubusercontent.com/44758056/163662547-6e3ef742-b965-4fc1-8dfa-ac188b059c34.jpg)

### v9 - Uncertainty Visible
![JPL - Asteroid Watch Page Desktop - v9 - Uncertainty Visible](https://user-images.githubusercontent.com/44758056/163662548-987a4509-583d-45c0-a479-5bba7a625ecb.jpg)

### v9 - Mobile
![JPL - Asteroid Watch Page Mobile - v9](https://user-images.githubusercontent.com/44758056/163662544-df7ad16f-5f0e-422b-8064-d92b430f6e76.jpg)

### Airplane Replacement
<img width="697" alt="asteroid watch airplane replacement" src="https://user-images.githubusercontent.com/44758056/163662603-dfea6e51-a057-47b9-bad2-739789875d87.png">

### Filters
![asteroid watch filters](https://user-images.githubusercontent.com/44758056/163662604-35e065a5-3f4b-4b64-b2c9-03a4100ecea0.jpg)

**Comment by stephiescastle** [comment]
Examples of the external header and footer have since been added, but we do not provide guidance.
